app/src/main/java/com/kotlin/readyplayerme/WebViewActivity.kt

```diff
@@ -126,29 +125,47 @@ class WebViewActivity : AppCompatActivity() {
     private fun handleAvatarCreated() {
         with(binding.webview){
             evaluateJavascript("""
-                window.addEventListener("message", receiveMessage, false)
-                function receiveMessage(event){                        
-                    // url display, used for displaying any string that starts with "https:"                    
-                    if (typeof data === "string" && event.data.indexOf("https:") !== -1) {
+                function subscribe(event) {
+                    // post message v1, this will be deprecated
+                    if(event.data.endsWith('.glb')) {
                         var content = document.querySelector(".content")
                         content.remove()
-                        MyScript.openDialog(event.data)
+                        WebView.receiveData(event.data)
                     }
-                    else{
-                        // catches only RPM related messages, displays stickers object
-                        // parse incoming message into JSON, and if object has readyPlayerMe and sticker
-                        // fields take stickers and print it as a string
-                          
-                        const data = JSON.parse(event.data)
+                    // post message v2
+                    else {
+                        const json = parse(event);
```

Comment:
When the event is an object, and not a text vaue that has passed JSON.stringify this parsing won't work.
I think we should handle this in the method.

```javascript
function parse(event) {
  if (typeof event === 'string') {
    return JSON.parse(event);
  }
  
  return event;
}
```
or dont need the function at all
```javascript
const json = typeof event === 'string' 
  ? JSON.parse(event)
  : event;
```

I'm just remembering that in some case we got a string object and in another case we got an actual object.

Reply:
I noticed it always lands as an object here in Kotlin, don't know why though. But type check makes more sense, updating it.